pkg/specgen/generate/storage.go

```diff
@@ -37,9 +37,11 @@ func finalizeMounts(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Ru
 	// Supersede from --volumes-from.
 	for dest, mount := range volFromMounts {
 		baseMounts[dest] = mount
+		delete(baseVolumes, dest)
```

Comment:
This needs a comment, it is completely non obvious why it is done like that.
The commit message is great but git history will be hard to track down once the code is moved for example so I like to see this in a comment.